### PR TITLE
Docker build and push images using circleci

### PR DESCRIPTION
**Pull request by jevonearth**
Also vendored dependencies using `dep`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - image: circleci/golang:1.11
+    working_directory: /go/src/github.com/ecadlabs/tezos_exporter
+
+    steps:
+    - checkout
+    - setup_remote_docker
+    - attach_workspace:
+        at: .
+    - run: docker build -t ecadlabs/tezos_exporter:$CIRCLE_BRANCH -t ecadlabs/tezos_exporter:${CIRCLE_SHA1:0:8} .
+    - run: docker images
```
Comment (juliusv) [inline .circleci/config.yml]: Is this step here just for debugging the log output? Should it still be there?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - docker_hub_release_tags:
+        requires:
+          - build
+        filters:
+          tags:
+            only: /^v[0-9]+(\.[0-9]+){2}(-.+|[^-.]*)$/
+          branches:
+            ignore: /.*/
+
+
```
Comment (juliusv) [inline .circleci/config.yml]: Extra empty lines at end of file.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - build:
+        filters:
+          tags:
+            only: /.*/
+    - docker_hub_branch:
+        requires:
+          - build
+        filters:
+          branches:
+            only: /.*/
```
Comment (juliusv) [inline .circleci/config.yml]: Do I understand correctly that this will build Docker images for all branches, including ephemeral PR ones? Is that desired?

**Comment by juliusv** [comment]
👍 besides comments

File: .circleci/config.yml (hunk 3 above)
Comment (jevonearth) [reply .circleci/config.yml]: Yes, that's too permissive, left over from debugging CI. I'll tighten it up.

**Pull request merged**